### ISSUE TITLE
Include and enable in ZeroNet by default

### ISSUE THREAD

**Issue by filips123** [HelloZeroNet/Plugin-PeerMessage#4]
Why is this plugin not included and enabled in ZeroNet by default? It probably should be.

**Comment by purplesyringa** [comment]
@HelloZeroNet Another thread about this.

**Comment by krzysztof113** [comment]
Bump on this, it's a blocker for me too.

**Comment by purplesyringa** [comment]
@HelloZeroNet Is there anything I can do to get the plugin included?

**Comment by HelloZeroNet** [comment]
One of the reasons of the plugin system was improved decentralization of the development and the ownership of the running code of the client. This way the source code can be updated separately from the client and I don't have to review the code/changes.

**Comment by filips123** [comment]
@HelloZeroNet I still think that even if this are third-party plugins, they should be enabled by default. Some users don't/can't install new plugins and for those you will still have to provide fallback if you want that your site is usable.

So, would it be possible to have plugins from plugin manager that can be updated separately but included/bundled by default in ZeroNet?

**Comment by krzysztof113** [comment]
> One of the reasons of the plugin system was improved decentralization of the development and the ownership of the running code of the client. This way the source code can be updated separately from the client and I don't have to review the code/changes.

And what workflow would be in such situation? User visits site that requires plugin to work and will there be prompt to install, redirect to plugin manager or other solution?

**Comment by HelloZeroNet** [comment]
> And what workflow would be in such situation? User visits site that requires plugin to work and will there be prompt to install, redirect to plugin manager or other solution?

Yes, you can ask user to install a plugin from a site directory and manage installed plugins on http://127.0.0.1:43110/Plugins : https://zeronet.io/docs/site_development/zeroframe_api_reference/#pluginaddrequest

**Comment by purplesyringa** [comment]
@krzysztof113 PeerMessage official zite is at http://127.0.0.1:43110/1CeEXxqemr5CcVQAAmrW13QYwZV4kAkQz6 (or peermessage.zn if you're using Name.YO).

**Comment by krzysztof113** [comment]
The reason I'm asking, is that here:
http://127.0.0.1:43110/1BTZh5pymEKzMYr3qgDtgr4dMmap77QvEs/?/about
only says to download the plugin and place the files self. 
I guess instruction is outdated (@krixano )

**Comment by imachug** [comment]
Huge sorry for such a long fix. I've pushed it and also updated peermessage.zn. Please update your plugins using http://127.0.0.1:43110/Plugins .

**Comment by purplesyringa** [comment]
Sorry

**Comment by caryoscelus** [comment]
i'm considering adding PeerMessage into `zeronet-conservancy` . any recommendations , warnings , anything to note ?

**Comment by purplesyringa** [comment]
It leaks IPs by default to trace the path messages travel through, you might want to patch that (see `trace` or `path`, I don't remember the exact name).
